### PR TITLE
docs: add readme and pull request template

### LINKED ISSUE #2: [Semaphore Boilerplate] Issue 2: Add the PR template and Readme files
The Semaphore-Stellar boilerplate needs an updated README file to include a clear explanation of how to use the boilerplate. Additionally, a standardized pull request (PR) template, similar to the format used in other repositories, should be created to streamline the contribution process and ensure consistency across PRs.

### PR DESCRIPTION
**Pull request by salazarsebas**
# Pull Request Overview

## 📝 Summary
This Pull Request consists in adding a readme where some terms are explained and facilitates the navigation between Semaphore Stellar projects. It also adds a template for Pull Requests.

## 🪧 Related Issues
- Closes #2 

## 🛠️ Type of Change
Mark with an x all the checkboxes that apply (like [x]).

- [x] 📝 Documentation (updates to README, docs, or comments)
- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] 👌 Enhancement (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)

## 🔄 Changes Made
The readme was changed and the rest was added from scratch.

## 📜 Checklist
Please check all the boxes that apply to this PR.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## 🔜 Next Steps
Follow the indications of the contributors and update as the project grows.

### CODE REVIEW

**Comment by salazarsebas** [comment]
Please tell me what you think of the pr. @evgongora @bitfalt  ⛩️🚀

File: .github/PR_TEMPLATE/PR_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+- [ ] My code follows the style guidelines of this project
+- [ ] I have performed a self-review of my own code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] I have made corresponding changes to the documentation
+- [ ] My changes generate no new warnings
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] New and existing unit tests pass locally with my changes
+- [ ] Any dependent changes have been merged and published in downstream modules
+
+## 🔜 Next Steps
```
Comment (bitfalt) [inline .github/PR_TEMPLATE/PR_TEMPLATE.md]: I think the Next Steps section isn't really necessary, the rest looks good overall!

**Review by bitfalt — CHANGES_REQUESTED** [review]
Just address the comment and everything looks good to me! Great job :D

File: .github/PR_TEMPLATE/PR_TEMPLATE.md (hunk 1 above)
Comment (salazarsebas) [reply .github/PR_TEMPLATE/PR_TEMPLATE.md]: Okay @bitfalt , I will remove that part.

**Comment by evgongora** [comment]
Awesome job! @salazarsebas

**Comment by salazarsebas** [comment]
> Awesome job! @salazarsebas

Ty @evgongora 🫡
I'm pleased to contribute to this project

**Review by bitfalt — APPROVED** [review]

**Comment by bitfalt** [comment]
Looks good! However, slight issue with the commits since they are not signed we can not merge the PR.

Check the following guide to have verified commits:
https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#gpg-commit-signature-verification

We'll change the rule for your PR, however we'd appreciate you follow the guide so that for future contributions you have the signed commits :D

**Comment by salazarsebas** [comment]
> Looks good! However, slight issue with the commits since they are not signed we can not merge the PR.
> 
> Check the following guide to have verified commits: https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#gpg-commit-signature-verification
> 
> We'll change the rule for your PR, however we'd appreciate you follow the guide so that for future contributions you have the signed commits :D

Thank you for sharing the guide with me and literally letting me know it existed. After a while I managed to set it up and now my commits are going to be signed. 🫡

**Pull request merged**